presto-hive/src/main/java/io/prestosql/plugin/hive/ViewReaderUtil.java

```diff
@@ -0,0 +1,198 @@
+    }
+
+    public static String encodeViewData(ConnectorViewDefinition definition)
+    {
+        byte[] bytes = VIEW_CODEC.toJsonBytes(definition);
+        String data = Base64.getEncoder().encodeToString(bytes);
+        return VIEW_PREFIX + data + VIEW_SUFFIX;
+    }
+
+    /**
+     * Supports decoding of Presto views
+     */
+    public static class PrestoViewReader
+            implements ViewReader
+    {
+        @Override
+        public ConnectorViewDefinition decodeViewData(String viewData, Table table, CatalogName catalogName)
+        {
+            checkCondition(viewData.startsWith(VIEW_PREFIX), HIVE_INVALID_VIEW_DATA, "View data missing prefix: %s", viewData);
+            checkCondition(viewData.endsWith(VIEW_SUFFIX), HIVE_INVALID_VIEW_DATA, "View data missing suffix: %s", viewData);
+            viewData = viewData.substring(VIEW_PREFIX.length());
+            viewData = viewData.substring(0, viewData.length() - VIEW_SUFFIX.length());
+            byte[] bytes = Base64.getDecoder().decode(viewData);
+            return VIEW_CODEC.fromJson(bytes);
+        }
+    }
+
+    /**
+     * Class to decode Hive view definitions
+     */
+    public static class HiveViewReader
+            implements ViewReader
+    {
+        private final HiveMetastoreClient metastoreClient;
+        private final TypeManager typeManager;
+
+        public HiveViewReader(HiveMetastoreClient hiveMetastoreClient, TypeManager typemanager)
+        {
+            this.metastoreClient = requireNonNull(hiveMetastoreClient, "metastoreClient is null");
+            this.typeManager = requireNonNull(typemanager, "typeManager is null");
+        }
+
+        @Override
+        public ConnectorViewDefinition decodeViewData(String viewSql, Table table, CatalogName catalogName)
+        {
+            try {
+                HiveToRelConverter hiveToRelConverter = HiveToRelConverter.create(metastoreClient);
+                RelNode rel = hiveToRelConverter.convertView(table.getDatabaseName(), table.getTableName());
+                RelToPrestoConverter rel2Presto = new RelToPrestoConverter();
+                String prestoSql = rel2Presto.convert(rel);
+                RelDataType rowType = rel.getRowType();
+                List<ViewColumn> columns = rowType.getFieldList().stream()
+                        .map(field -> new ViewColumn(
+                                field.getName(),
+                                typeManager.fromSqlType(getTypeString(field.getType())).getTypeId()))
+                        .collect(toImmutableList());
+                return new ConnectorViewDefinition(prestoSql,
+                        Optional.of(catalogName.toString()),
+                        Optional.of(table.getDatabaseName()),
+                        columns,
+                        Optional.ofNullable(table.getParameters().get(TABLE_COMMENT)),
+                        Optional.empty(),
+                        true);
```

Comment:
Just double checking. Is this deliberate choice to execute Hive views by INVOKER? Previously they were executed by INVOKER. It sounds like a backward incompatible change (just noting for release notes if it is deliberate choice).

Reply:
@kokosing Thanks for noticing that. I just noticed it is incompatible with previous version.  It has been running in our production environment with RunAsInvoker set to true. But if it is preferred to be backward compatible, I am ok to change.

Reply:
I am not that much concerned about being backward compatible, I wonder more about correctness. The question is what is the view semantic in Hive. Does Hive support running as invoker only or as owner only, does it allow to choose? We should do what view says we should do.

I found it with a case when user has only access to a view and no access to underlying table, and query fails because we check the access to the table for that user (invoker), while previously we checked access to table but for the view owner.

Do you use views in your production environment for access control purposes?

Regarding backward compatibility, I hope that previous behavior was deliberate but I am not 100% sure.

Reply:
Actually, I tested this with Hive and it works there, while in Presto I now get access denied.

Reply:
https://github.com/prestosql/presto/pull/5597

Reply:
@findepi - Would it be better to have this Config controlled?

Reply:
@dmitryfill
In https://github.com/prestosql/presto/pull/5597 we made those VIEWs back in line with SQL standard.
i do not see why we would want this to be controlled by config just yet. 
Please do not hesitate to file a new issue and ideally provide some rationale too.

Reply:
Alternatively, possibly more powerful would be to have `ALTER VIEW x SET SECURITY DEFINER`

Reply:
I was too quick. It won't work for Hive views as Presto do not control them.